snippets/LimitedWordTextArea.md

```diff
@@ -0,0 +1,62 @@
+### LimitedWordTextArea
+
+Renders a textarea component with a word limit.
+
+Use the value of the `value` prop to determine the initial `state.content` and `state.wordCount` and the value of the `limit` props to determine the value of `state.limit`.
+Create a method, `handleChange`, which trims the `event.target.value` data if necessary and uses `Component.prototype.setState` to update `state.content` and `state.wordCount`, and bind it to the component's context.
+In the`render()` method, use a`<div>` to wrap both the`<textarea>` and the `<p>` element that displays the character count and bind the `onChange` event of the `<textarea>` to the `handleChange` method.
+
+```jsx
+class LimitedWordTextArea extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      content: props.value,
+      wordCount: props.value.split(' ').filter(Boolean).length,
+      limit: props.limit
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    let newContent = event.target.value;
+
+    if (newContent.split(' ').filter(Boolean).length > this.state.limit) {
+      newContent = newContent
```

Comment:
Since this should be used in the constructor as well we should move this to a static or an instance method.